rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+    if (!value_name.IsTypedArray()) {                                          \
+        Napi::TypeError::New(                                                  \
+            env, "BLST_ERROR: " js_name " must be a BlstBuffer")               \
+            .ThrowAsJavaScriptException();                                     \
+        m_has_error = true;                                                    \
+        return;                                                                \
+    }                                                                          \
+    Napi::TypedArray arr_name##_array = value_name.As<Napi::TypedArray>();     \
+    if (arr_name##_array.TypedArrayType() != napi_uint8_array) {               \
+        Napi::TypeError::New(                                                  \
+            env, "BLST_ERROR: " js_name " must be a BlstBuffer")               \
+            .ThrowAsJavaScriptException();                                     \
+        m_has_error = true;                                                    \
+        return;                                                                \
+    }                                                                          \
+    Napi::Uint8Array arr_name =                                                \
+        arr_name##_array.As<Napi::TypedArrayOf<uint8_t>>();
+
+class BlstTsAddon;
+
+typedef enum { Affine, Jacobian } CoordType;
+
+/**
+ * Checks a byte array to see if it is all zeros. Can pass start byte for the
+ * cases where the first byte is the tag (infinity point and
+ * compress/uncompressed).
+ *
+ * @param data uint8_t*
+ * @param start_byte size_t
+ * @param byte_length size_t
+ *
+ * @return bool
+ */
+bool is_zero_bytes(
+    const uint8_t *data, const size_t start_byte, const size_t byte_length);
+
+/**
+ * Checks if a byte array is a valid length. If not, sets the error message and
+ * returns false.  If valid returns true for use in conditional statements.
+ *
+ * @param[out] error_out &std::string - error message to set if invalid
+ * @param[in] byte_length size_t - length of the byte array to validate
+ * @param[in] length1 size_t - first valid length
+ * @param[in] length2 size_t - second valid length (optional)
+ *
+ * @return bool
+ */
+bool is_valid_length(
+    std::string &error_out,
+    size_t byte_length,
+    size_t length1,
+    size_t length2 = 0);
+
+/**
+ * Circular dependency if these are moved up to the top of the file.
+ */
+#include "functions.h"
+#include "public_key.h"
+#include "secret_key.h"
+#include "signature.h"
+
+/**
+ * BlstTsAddon is the main entry point for the library. It is responsible
+ * for initialization and holding global values.
+ */
+class BlstTsAddon : public Napi::Addon<BlstTsAddon> {
+   public:
+    std::string _dst;
+    std::string _blst_error_strings[8];
+    Napi::FunctionReference _secret_key_ctr;
+    Napi::FunctionReference _public_key_ctr;
+    Napi::FunctionReference _signature_ctr;
```

Comment:
The `_` prefix generally denotes the member is private. Do they need to be public here? I'd recommend encapsulating it if possible. Otherwise, users may break some invariants by modifying the internals of the instance.

Reply:
Updated.